### PR TITLE
Update 2.14 with rancher changes

### PR DESCRIPTION
**Pull request by dramich**
This will rename the binaries created to `rancher-helm` and `rancher-tiller` along with adding rancher required things. 

Rancher will needs some updates to use this new version:
1. binary name changes
2. Probe flag updated to `probe-listen` for tiller. This is now builtin to tiller.

### CODE REVIEW

File: pkg/releasetesting/environment_test.go [hunk 1]
```diff
@@ -178,6 +178,6 @@ func newCreateFailingKubeClient() *createFailingKubeClient {
 	}
 }
 
-func (p *createFailingKubeClient) Create(ns string, r io.Reader, t int64, shouldWait bool) error {
+func (p *createFailingKubeClient) Create(name, ns string, r io.Reader, t int64, shouldWait bool) error {
 	return errors.New("We ran out of budget and couldn't create finding-nemo")
```
Comment (daxmc99) [inline pkg/releasetesting/environment_test.go]: 👀

File: cmd/tiller/tiller.go [hunk 2]
```diff
@@ -215,26 +225,30 @@ func start() {
 
 	srvErrCh := make(chan error)
 	probeErrCh := make(chan error)
-	go func() {
+	done := make(chan struct{})
```
Comment (daxmc99) [inline cmd/tiller/tiller.go]: Is this tiller upstream or a change needed for rancher? also what does it do?

File: pkg/kube/client.go [hunk 3]
```diff
@@ -111,6 +111,28 @@ func (c *Client) Create(namespace string, reader io.Reader, timeout int64, shoul
...
+		// if resource is created under a namespace we have to make sure the namespace matches
+		if info.Namespace != namespace && info.Namespace != "" {
+			return fmt.Errorf("resource's namespace %s doesn't match the current namespace %s", info.Namespace, namespace)
+		}
+		if name != "" {
+			raw := info.Object.(runtime.Unstructured).UnstructuredContent()
+			label, ok := GetValue(raw, "metadata", "labels")
+			if ok {
+				if v, ok := label.(map[string]interface{}); ok {
+					v["io.cattle.field/appId"] = name
```
Comment (luthermonson) [inline pkg/kube/client.go]: worth making "io.cattle.field/appId" a constant? you use it 4 times could be helpful

File: cmd/tiller/tiller.go (hunk 2 above)
Comment (dramich) [reply cmd/tiller/tiller.go]: It's ours. Stops a race condition with registering before we register...

File: pkg/kube/client.go (hunk 3 above)
Comment (dramich) [reply pkg/kube/client.go]: Done.

File: cmd/tiller/tiller.go (hunk 2 above)
Comment (luthermonson) [reply cmd/tiller/tiller.go]: if youre going to gate the second go routine from running until the first is done why not just combine into one or call the second one from the first?

File: pkg/kube/client.go (hunk 3 above)
Comment (luthermonson) [reply pkg/kube/client.go]: could you combine this logic some how with createResourceWithAppID? feels like similar logic

**Comment by cjellick** [comment]
LGTM

**Pull request merged**